src/OpenApi/gen/XmlCommentGenerator.Parser.cs

```diff
@@ -0,0 +1,172 @@
+                method.GetDocumentationCommentXml(CultureInfo.InvariantCulture, expandIncludes: true, cancellationToken: cancellationToken) is string xml)
+            {
+                comments.Add((name, xml));
+            }
+        }
+        return comments;
+    }
+
+    // Type names are used in a `typeof()` expression, so we need to replace generic arguments
+    // with empty strings to avoid compiler errors.
+    private static string ReplaceGenericArguments(string typeName)
+    {
+        var stack = new Stack<int>();
+        var result = new StringBuilder(typeName);
+
+        for (var i = 0; i < result.Length; i++)
+        {
+            if (result[i] == '<')
+            {
+                stack.Push(i);
+            }
+            else if (result[i] == '>' && stack.Count > 0)
+            {
+                var start = stack.Pop();
+                // Replace everything between < and > with empty strings separated by commas
+                var segment = result.ToString(start + 1, i - start - 1);
+                var commaCount = segment.Count(c => c == ',');
+                var replacement = string.Join(",", Enumerable.Repeat("", commaCount + 1));
+                result.Remove(start + 1, i - start - 1);
+                result.Insert(start + 1, replacement);
+                i = start + replacement.Length + 1;
+            }
+        }
+
+        return result.ToString();
+    }
+
+    internal static IEnumerable<(string, string?, XmlComment?)> ParseComments(
+        (List<(string, string)> RawComments, Compilation Compilation) input,
+        CancellationToken cancellationToken)
+    {
+        var compilation = input.Compilation;
+        var comments = new List<(string, string?, XmlComment?)>();
+        foreach (var (name, value) in input.RawComments)
+        {
+            if (DocumentationCommentId.GetFirstSymbolForDeclarationId(name, compilation) is ISymbol symbol)
+            {
+                var parsedComment = XmlComment.Parse(symbol, compilation, value, cancellationToken);
+                if (parsedComment is not null)
+                {
+                    var typeInfo = symbol is IPropertySymbol or IMethodSymbol
+                    ? ReplaceGenericArguments(symbol.ContainingType.OriginalDefinition.ToDisplayString(_typeKeyFormat))
+                    : ReplaceGenericArguments(symbol.OriginalDefinition.ToDisplayString(_typeKeyFormat));
+                    var propertyInfo = symbol is IPropertySymbol or IMethodSymbol
+                        ? symbol.Name
+                        : null;
+                    comments.Add((typeInfo, propertyInfo, parsedComment));
+                }
+            }
+        }
+        return comments;
+    }
+
+    internal static bool FilterInvocations(SyntaxNode node, CancellationToken _)
+        => node is InvocationExpressionSyntax { Expression: MemberAccessExpressionSyntax { Name.Identifier.ValueText: "AddOpenApi" } };
```

Comment:
Is this filtering down to the assembly/namespace somewhere? Just thinking about if someone defines their own `AddOpenApi()` overload for convenience in their codebase where they use the same namespace for convenience - wouldn't want the earlier code throwing because it's an unexpected overload.

Reply:
We're operating on the syntactic model at this phase, which doesn't give us access to the semantic model to resolve the containing assembly of the method.

You make a great point about `AddOpenApi` being a method name that is ripe for overload so perhaps we should decrease the severity of the failure mode from an error to a log message.

Reply:
I ended up adding an explicit check on the containing assembly later and removed the exception in favor of a no-op.